### PR TITLE
refactor: consider user role staff for masquerading

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
### Description
This PR considers the `staff` course role as well so masquerading as a course staff works too. I'm trying to figure out what causes this behavior. To reproduce:
1. As a course admin (instructor), go to your course with this xblock installed
2. Enter the unit
3. Even though you're viewing the course as staff, select view course as `staff`. 

The `Grade Submissions` should disappear after this. What happens is that when you enter the course you have the `instructor` role, but then you view the course as staff (after pressing view as staff) now you have the `staff` role. So even though you reload the page or enter the course again, the `Grade Submissions` button won't appear again unless you log out your account and login again.

### CODE REVIEW

**Comment by BryanttV** [comment]
@mariajgrimaldi When I select view course as staff, and click the `Grade Submissions` button, I get that I don't have permission. Should that be the behavior?

```bash
  File "/openedx/requirements/xblock-mindmap/mindmap/mindmap.py", line 472, in get_instructor_grading_data
    require(self.is_instructor())
  File "/openedx/requirements/xblock-mindmap/mindmap/mindmap.py", line 778, in require
    raise PermissionDenied
django.core.exceptions.PermissionDenied
```

**Comment by mariajgrimaldi** [comment]
@BryanttV: no, thanks. I'll make sure to fix it right away.

**Comment by mariajgrimaldi** [comment]
@BryanttV: can you test again? Thanks!

**Comment by BryanttV** [comment]
Perfect, it works!

File: mindmap/mindmap.py [hunk 1]
```diff
@@ -589,7 +589,7 @@ def remove_grade(self, data, _suffix="") -> dict:
         # Lazy import: import here to avoid app not ready errors
         from submissions.api import reset_score  # pylint: disable=import-outside-toplevel
 
-        require(self.is_instructor())
+        require(self.is_instructor() or self.is_course_staff(self.get_current_user()))
```
Comment (Ian2012) [inline mindmap/mindmap.py:592]: Can you make this a property?

Comment (mariajgrimaldi) [reply mindmap/mindmap.py:592]: Yeah. But I'll need to do some refactoring first

Comment (mariajgrimaldi) [reply mindmap/mindmap.py]: Done @Ian2012. Can you check again?

**Comment by mariajgrimaldi** [comment]
There have been some changes after you @BryanttV reviewed, can you check again? Thanks!

**Review by BryanttV — APPROVED** [review]

**Review by Ian2012 — APPROVED** [review]

**Pull request merged**